### PR TITLE
Fix a subtle special case of the shutdown hanger.

### PR DESCRIPTION
**Pull request by neunhoef**

### CODE REVIEW

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6498/

File: arangod/Utils/CollectionKeysRepository.h [hunk 1]
```diff
@@ -108,6 +118,13 @@ class CollectionKeysRepository {
   //////////////////////////////////////////////////////////////////////////////
 
   static size_t const MaxCollectCount;
+
+  ////////////////////////////////////////////////////////////////////////////
+  /// @brief stopped flag, indicating that no more CollectionKeys can be
+  /// stored
+  ////////////////////////////////////////////////////////////////////////////
+
+  bool _stopped;
```
Comment (jsteemann) [inline arangod/Utils/CollectionKeysRepository.h]: this variable is never initialized, so its initial value is UB.
```suggestion
  bool _stopped = false;
```

**Comment by neunhoef** [comment]
Note that I checked that `CollectionKeysRepository::store` is called in only a single place, which is `arangod/MMFiles/MMFilesRestReplicationHandler.cpp:667`. And there the RestHandler immediately returns. Therefore I concluded that it OK if during shutdown the object is silently not moved into the registry. Other places will have to be prepared to not find it there anyway because of the timeouts.

File: arangod/Utils/CollectionKeysRepository.h (hunk 1 above)
Comment (neunhoef) [reply arangod/Utils/CollectionKeysRepository.h]: This is an oversight, I will fix immediately!

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6499/

**Pull request merged**